### DISCUSSION TITLE
Expo cannot open development build on android emulator

### DISCUSSION THREAD

**Discussion by hayata-suenaga** [expo/expo#18893] [Archived: Questions]
I'm trying to run my Expo managed project on Android simulator with `npx expo run:android`.

I have `expo-dev-client` installed.

When I run `npx expo run:android`, it fails with the message:

```
CommandError: Couldn't open Android app with activity "com.my.app.name://expo-development-client/?url=http%3A%2F%2F192.168.0.13%3A8081" on device "Pixel_4".
The app might not be installed, try installing it with: npx expo run:android -d Pixel_4
```

I ran `npx expo run:android -d Pixel_4` as suggested in the above error message only to get the exact same error message as above.

On the emulator, I was able to confirm that the development build is installed.

However, the Expo CLI apparently cannot open the app.

Could someone guide me to the right direction to solve this issue?

**Comment by AxanIqbal** [comment]
having the same issue does anyone able to solve this?

**Comment by lindan4** [comment]
Did you find a solution to this?

**Reply by AxanIqbal** [reply]
Well yes, after running expo from global one works for me not npx expo or yarn android the cli. after expo 46 in package cli never works

expo run:android

**Reply by lindan4** [reply]
What do you mean by global one? Do you mean one of 'yarn start' or 'expo start --dev-client'?

**Reply by AxanIqbal** [reply]
Just run `expo run:android`

**Reply by lindan4** [reply]
What version of Expo are you using? I'm using version 47.

**Reply by AxanIqbal** [reply]
My expo version in project is 47.0.10 and global is 6.0.8

**Comment by RRed** [comment]
I fixed this issue by clearing the simulator. i.e. open the simulator and, under the device, clear content and settings. Also, you can try using the "--clean" flag.

**Comment by TahirMehmood8082** [comment]
expo run:android 
solved my issue

**Comment by dbaltor** [comment]
I've just faced the same issue in an with an android emulator with Android Studio 2025.1.2 after upgrading Expo but I managed to fix the issue thanks to @RRed's tip above: just reset the emulator by wipping all data out. 
**- Open Virtual Device Manager -> Click on the three dots at the end of the emulator line -> Wipe Data.**

**Comment by leungtinyau27m5** [comment]
I was facing this issue with `yarn android --device`, and the error was `The emulator (Pixel_XL) quit before it finished opening. You can try starting the emulator manually from the terminal with`.  I tried to kill the adb server, and run the emulator with `emualtor @{avd_name}`, but got no luck.

However the app is successfully installed and works fine on emulator with `npm run android -- --device`

i think it might be the problem of `yarn`

```json
"expo": "~52.0.46"
"packageManger": "yarn@4.9.1"
```
